hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,225 @@
+Typically each transaction consists of:
+1. A START signal, issued by host.
+1. An address, issued by host, encoded as 7 or 10 bits.
+1. A R/W bit indicating whether the transaction is a read from the target device, or a write to the target device.
+The R/W bit is encoded along with the address. 
+1. An Acknowledge signal (ACK) sent by the target device. 
+1. Data bytes, where the number of bytes required is indicated by the host,
+in a manner which differs between reads and writes.
+    - For write transactions, the target device sends an ACK signal after every byte received.
+    The host indicates the end of a transaction by sending a STOP or RESTART signal.
+    - For read transactions, the device continues to send data as long as the host acknowledges the target-issued data by sending an ACK signal.
+    Once the host has received all the required data it indicates that no more data is needed by explicitly de-asserting the ACK signal (this is called a NACK signal) just before sending a STOP or RESTART signal. 
+1. A STOP signal or a RESTART signal.
+
+This protocol is generally quite flexible with respect to timing constraints, and slow enough to be managed by a software microcontroller, however such an implementation requires frequent activity on the part of the microcontroller.
+This IP presents a simple register interface and state-machine to manage the corresponding I/O pins directly using a byte-formatted programming model.
+
+## Compatibility
+
+This IP block should be compatible with any device covered by the [I2C specification](https://www.nxp.com/docs/en/user-guide/UM10204.pdf), operating at speeds up to 1 Mbaud.
+In order to support any device, this IP issues addresses in 7-bit encoding whenever possible, as not all devices may be able to support 10-bit encoding.
+(It remains the obligation of system designers to ensure that devices which cannot support 10-bit encoding remain in a 7-bit address space.)
+This IP also supports clock-stretching, should that be required by remote devices.
+
+# Theory of Operations 
+
+## Block Diagram
+
+![](I2C_block_diagram.svg)
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/i2c/data/i2c.hjson" >}}
+
+## Design Details 
+
+### Override Mode for Direct Pin Access
+
+The I2C hardware interface consists of two external pins, SCL and SDA, whose behavior is described in the [I2C specification](https://www.nxp.com/docs/en/user-guide/UM10204.pdf).
+These pins are typically controlled by an internal state machine.
+However, there is a simpler "override" mode, by which these pins can be directly addressed by software.
+This override mode is useful for troubleshooting or error-recovery.
+
+To enter override mode, the register field {{< regref OVRD.TXOVRDEN >}} is asserted by software.
+In this state the output drivers scl_tx_o and sda_tx_o are controlled directly by the register fields {{< regref OVRD.SCLVAL >}} () and {{< regref OVRD.SDAVAL >}}.
```

Comment:
`backticks` around `scl_tx_o` and `sda_tx_o` perhaps.

Intentional empty () parenthesis?